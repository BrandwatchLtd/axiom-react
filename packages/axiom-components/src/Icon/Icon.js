import PropTypes from "prop-types";
import React, { Component } from "react";
import classnames from "classnames";
import { icons } from "@brandwatch/axiom-materials";
import Base from "../Base/Base";

import "./Icon.css";

export default class Icon extends Component {
  static propTypes = {
    /** @ignore */
    className: PropTypes.string,
    /** Text color styling */
    color: PropTypes.oneOf([
      "body",
      "day",
      "night",
      "error",
      "success",
      "warning",
      "disabled",
      "subtle",
      "twitter",
      "facebook",
      "instagram",
      "forbidden-planet",
      "tiny-clanger",
      "critical-mass",
      "fantastic-voyage",
      "paradise-lost",
      "serene-sea",
      "event-horizon",
      "electric-dreams",
      "outer-limits",
      "giant-leap",
      "moon-lagoon",
      "space-invader",
      "extraterrestrial",
      "terra-form",
      "primeval-soup",
      "future-shock",
      "sun-maker",
      "new-horizon",
      "blast-off",
      "crash-course",
      "solar-rust",
      "ground-control",
      "space-oddity",
      "rocky-planet",
      "deep-thought",
      "luna-dust",
      "sentiment-negative",
      "sentiment-positive",
      "sentiment-neutral",
      "ui-accent",
    ]),
    /**
     * Inline styling that allows the Icon to placed next to other inline
     * elements or text.
     */
    /** Height of icon (with unit). */
    height: PropTypes.string,
    inline: PropTypes.bool,
    /** Name of the icon that will be displayed */
    name: PropTypes.oneOf([
      "annotate",
      "analytics-alert",
      "arrow-down",
      "arrow-down-left",
      "arrow-down-right",
      "arrow-left",
      "arrow-right",
      "arrow-up",
      "arrow-up-left",
      "arrow-up-right",
      "at",
      "bar-chart",
      "benchmark",
      "box-collapse",
      "box-expand",
      "brandwatch",
      "chevron-down",
      "chevron-left",
      "chevron-right",
      "chevron-up",
      "clock",
      "cloud-upload",
      "copy",
      "cross",
      "customise",
      "data",
      "deck",
      "dot",
      "dot-outline",
      "double-chevron-left",
      "double-chevron-right",
      "download",
      "drag",
      "duplicate",
      "ellipsis",
      "email",
      "exclude",
      "facebook",
      "feedback",
      "filter",
      "filter-line",
      "flip-horizontal",
      "flip-vertical",
      "folder",
      "folder-closed",
      "followers",
      "form-resize",
      "format-bold",
      "format-italic",
      "format-quote",
      "format-title",
      "format-underline",
      "grid",
      "growth",
      "hamburger",
      "heart",
      "home",
      "information",
      "input",
      "information-circle",
      "installation",
      "instagram",
      "integrations",
      "iris",
      "kred",
      "line-graph",
      "link",
      "list",
      "location",
      "lock",
      "log",
      "magnify-glass",
      "mask",
      "minus",
      "monitor",
      "news",
      "blog",
      "moon",
      "globe",
      "review",
      "blog",
      "open",
      "pause",
      "pencil",
      "pinterest",
      "plugin",
      "plus",
      "speech",
      "preview",
      "question-mark",
      "question-mark-circle",
      "reddit",
      "refresh",
      "replace",
      "reply",
      "reports",
      "retweet",
      "ring",
      "settings",
      "share",
      "shuffle",
      "social-panel",
      "star",
      "sun",
      "switch",
      "tag",
      "theme",
      "tick",
      "tools",
      "trash",
      "triangle-down",
      "triangle-left",
      "triangle-right",
      "triangle-up",
      "twitter",
      "twitter-mono",
      "unlock",
      "user",
      "warning",
      "warning-circle",
      "twitter",
      "facebook",
      "instagram",
      "reddit",
      "pinterest",
      "youtube",
      "qq",
      "tumblr",
      "4chan",
      "sina-weibo",
      "lexis-nexis",
      "twitter-mono",
      "facebook-mono",
      "instagram-mono",
      "reddit-mono",
      "pinterest-mono",
      "youtube-mono",
      "qq-mono",
      "tumblr-mono",
      "4chan-mono",
      "sina-weibo-mono",
      "lexis-nexis-mono",
      "source",
      "vk",
      "vk-mono",
    ]).isRequired,
    /** Size of icon (with unit). Not used when height or width is provided */
    size: PropTypes.string,
    /** Spacing applied to the left of the Icon. Must be used with the inline property */
    spaceLeft: PropTypes.oneOf(["x1", "x2"]),
    /** Spacing applied to the right of the Icon. Must be used with the inline property */
    spaceRight: PropTypes.oneOf(["x1", "x2"]),
    /** Style */
    style: PropTypes.object,
    /** Width of icon (with unit). */
    width: PropTypes.string,
  };

  static defaultProps = {
    size: "1rem",
  };

  render() {
    const {
      color,
      className,
      inline,
      name,
      size,
      width,
      height,
      spaceLeft,
      spaceRight,
      style = {},
      ...rest
    } = this.props;
    const sizeStyle =
      height || width ? { width, height } : { width: size, height: size };

    const classes = classnames(className, "ax-icon", `ax-icon--${name}`, {
      "ax-icon--inline": inline,
      [`ax-icon--space-left-${spaceLeft}`]: spaceLeft,
      [`ax-icon--space-right-${spaceRight}`]: spaceRight,
    });

    if (!icons[name]) {
      return null;
    }

    const { body, viewBox } = icons[name];

    return (
      <Base
        textColor={color}
        {...rest}
        Component="svg"
        className={classes}
        dangerouslySetInnerHTML={{ __html: body }}
        style={{
          ...sizeStyle,
          ...style,
        }}
        viewBox={viewBox}
      />
    );
  }
}
