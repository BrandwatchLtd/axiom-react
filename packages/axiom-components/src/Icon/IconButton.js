import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Button from '../Button/Button';
import ButtonIcon from '../Button/ButtonIcon';

const sizeMap = {
  'small': '1rem',
  'medium': '2rem',
  'large': '3rem',
  'huge': '5rem',
};

export default class IconButton extends Component {
  static propTypes = {
    /** Name of the icon that will be displayed */
    name: PropTypes.string.isRequired,
    /** Size of the button and icon */
    size: PropTypes.oneOf(['small', 'medium', 'large', 'huge']),
  };

  static defaultProps = {
    size: 'medium',
  }

  render() {
    const { name, size, ...rest } = this.props;

    return (
      <Button style="secondary" { ...rest } shape="circle" size={ size }>
        <ButtonIcon name={ name } size={ sizeMap[size] } />
      </Button>
    );
  }
}
