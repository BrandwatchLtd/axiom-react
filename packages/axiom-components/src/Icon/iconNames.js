const iconNames = [
  "4chan",
  "4chan-mono",
  "annotate",
  "analytics-alert",
  "arrow-down",
  "arrow-down-left",
  "arrow-down-right",
  "arrow-up",
  "arrow-up-left",
  "arrow-up-right",
  "at",
  "bar-chart",
  "benchmark",
  "blog",
  "box-collapse",
  "box-expand",
  "chevron-down",
  "chevron-left",
  "chevron-right",
  "chevron-up",
  "clock",
  "cloud-upload",
  "cross",
  "customise",
  "data",
  "deck",
  "dot",
  "dot-outline",
  "double-chevron-left",
  "double-chevron-right",
  "download",
  "drag",
  "duplicate",
  "ellipsis",
  "email",
  "exclude",
  "facebook-mono",
  "facebook",
  "filter",
  "filter-line",
  "flip-horizontal",
  "flip-vertical",
  "folder",
  "folder-closed",
  "followers",
  "form-resize",
  "format-bold",
  "format-italic",
  "format-quote",
  "format-title",
  "format-underline",
  "globe",
  "grid",
  "growth",
  "hamburger",
  "heart",
  "home",
  "information",
  "input",
  "information-circle",
  "installation",
  "instagram-mono",
  "instagram",
  "integrations",
  "iris",
  "kred",
  "lexis-nexis",
  "lexis-nexis-mono",
  "line-graph",
  "link",
  "list",
  "location",
  "lock",
  "log",
  "magnify-glass",
  "minus",
  "monitor",
  "moon",
  "news",
  "open",
  "pause",
  "pencil",
  "pinterest-mono",
  "pinterest",
  "plugin",
  "plus",
  "preview",
  "qq-mono",
  "qq",
  "question-mark",
  "question-mark-circle",
  "reddit-mono",
  "reddit",
  "refresh",
  "replace",
  "reply",
  "reports",
  "retweet",
  "review",
  "ring",
  "settings",
  "share",
  "shuffle",
  "sina-weibo-mono",
  "sina-weibo",
  "social-panel",
  "source",
  "speech",
  "star",
  "sun",
  "switch",
  "tag",
  "theme",
  "tick",
  "tools",
  "trash",
  "triangle-down",
  "triangle-left",
  "triangle-right",
  "triangle-up",
  "tumblr-mono",
  "tumblr",
  "twitter-mono",
  "twitter",
  "unlock",
  "user",
  "warning",
  "warning-circle",
  "youtube-mono",
  "youtube",
];

export default iconNames;
