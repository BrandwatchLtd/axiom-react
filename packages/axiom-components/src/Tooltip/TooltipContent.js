import React, { Component } from 'react';
import ContextContent from '../Context/ContextContent';

export default class TooltipContent extends Component {
  render() {
    return (
      <ContextContent { ...this.props } padding="small" />
    );
  }
}
