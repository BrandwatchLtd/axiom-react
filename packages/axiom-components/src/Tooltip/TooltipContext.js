import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Context from '../Context/Context';

export default class TooltipContext extends Component {
  static propTypes = {
    color: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  }
  render() {

    const {
      color,
      ...rest
    } = this.props;

    return (
      <Context color={ color } { ...rest } />
    );
  }
}
