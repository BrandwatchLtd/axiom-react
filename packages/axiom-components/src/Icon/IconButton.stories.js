import React from "react";
import IconButton from "./IconButton";

export default {
  title: "IconButton",
  component: IconButton,
};

export function Default({ name, size, buttonStyle, iconColor }) {
  return (
    <IconButton
      name={name}
      size={size}
      buttonStyle={buttonStyle}
      iconColor={iconColor}
    />
  );
}

Default.args = { name: "cross" };

export function MenuOpen() {
  return <IconButton name="ellipsis" aria-label="open" />;
}
