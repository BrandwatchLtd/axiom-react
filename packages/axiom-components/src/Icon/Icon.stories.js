import React from 'react';
import Icon from './Icon';
import Grid from '../Grid/Grid';
import GridCell from '../Grid/GridCell';
import iconNames from './iconNames';

export default {
  title: 'Components/Icon',
  component: Icon,
};

export function Default() {
  return (
    <Grid responsive={ false }>
      { iconNames.map(name => (
        <GridCell key={ name } width={ 25 }>
          <Grid gutters="tiny" none responsive={ false } verticalAlign="middle">
            <GridCell>
              <Icon name={ name } />
            </GridCell>

            <GridCell>{ name }</GridCell>
          </Grid>
        </GridCell>
      )) }
    </Grid>
  );
}
