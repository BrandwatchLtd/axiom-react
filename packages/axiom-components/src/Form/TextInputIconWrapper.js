import classnames from "classnames";
import PropTypes from "prop-types";
import React, { Component } from "react";
import { isComponent } from "@brandwatch/axiom-utils";
import Icon from "../Icon/Icon";
import Link from "../Typography/Link";
import Tooltip from "../Tooltip/Tooltip";
import TooltipContent from "../Tooltip/TooltipContent";
import TooltipContext, { TooltipContextRef } from "../Tooltip/TooltipContext";
import TooltipSource from "../Tooltip/TooltipSource";
import TooltipTarget from "../Tooltip/TooltipTarget";

export const TextInputIconWrapperRef = "TextInputIconWrapper";

export default class TextInputIconWrapper extends Component {
  static propTypes = {
    /** Child element */
    children: PropTypes.node.isRequired,
    /** Alignment inside the input field */
    align: PropTypes.oneOf(["left", "right"]),
    /** Click handler, applies styling to indicate it is clickable. */
    onClick: PropTypes.func,
    /** Optional tooltip for the icon. */
    tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    /** Optional tooltip color. Default is "carbon" */
    tooltipColor: PropTypes.oneOf([
      "success",
      "warning",
      "error",
      "info",
      "carbon",
      "white",
    ]),
  };

  static defaultProps = {
    align: "right",
    tooltipColor: "carbon",
  };

  static typeRef = TextInputIconWrapperRef;

  render() {
    const {
      align,
      onClick,
      tooltip,
      children,
      tooltipColor,
      ...rest
    } = this.props;

    const className = classnames("ax-input__icon", {
      [`ax-input__icon--align-${align}`]: align,
    });

    const icon = tooltip ? (
      <Tooltip isVisible>
        <TooltipTarget>{children}</TooltipTarget>
        <TooltipSource width="auto">
          {isComponent(tooltip, TooltipContextRef) ? (
            tooltip
          ) : (
            <TooltipContext color={tooltipColor}>
              <TooltipContent size="tiny">{tooltip}</TooltipContent>
            </TooltipContext>
          )}
        </TooltipSource>
      </Tooltip>
    ) : (
      children
    );

    return (
      <span className={className}>
        {onClick ? (
          <Link {...rest} onClick={onClick} style="subtle">
            {icon}
          </Link>
        ) : (
          icon
        )}
      </span>
    );
  }
}
