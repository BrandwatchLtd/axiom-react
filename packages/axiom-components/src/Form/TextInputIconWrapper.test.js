import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import TextInputIconWrapper from "./TextInputIconWrapper";
import TooltipContext from "../Tooltip/TooltipContext";
import TooltipContent from "../Tooltip/TooltipContent";
import Icon from "../Icon/Icon";

const getComponent = (props) =>
  renderer.create(<TextInputIconWrapper {...props} />);

describe("TextInputIconWrapper", () => {
  let props;

  beforeEach(() => {
    props = {
      children: <Icon name="dot" />,
    };
  });

  it("renders with defaultProps", () => {
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with align="left"', () => {
    props.align = "left";
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with onClick", () => {
    props.onClick = () => {};
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with tooltip when string passed", () => {
    props.tooltip = "I am a tooltip";

    const component = getComponent(props);
    const tree = component.toJSON();

    const wrapper = shallow(<TextInputIconWrapper {...props} />);
    wrapper.simulate("mouseenter");

    expect(tree).toMatchSnapshot();
    expect(wrapper.find(TooltipContent)).toHaveLength(1);
  });

  it("renders with tooltip when TooltipContext passed", () => {
    props.tooltip = (
      <TooltipContext>
        <TooltipContent size="tiny">I am a tooltip</TooltipContent>
      </TooltipContext>
    );

    const component = getComponent(props);
    const tree = component.toJSON();

    const wrapper = shallow(<TextInputIconWrapper {...props} />);
    wrapper.simulate("mouseenter");

    expect(tree).toMatchSnapshot();
    expect(wrapper.find(TooltipContext)).toHaveLength(1);
  });
});
