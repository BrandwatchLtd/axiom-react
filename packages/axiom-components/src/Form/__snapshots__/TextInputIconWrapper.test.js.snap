// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputIconWrapper renders with align="left" 1`] = `
<span
  className="ax-input__icon ax-input__icon--align-left"
>
  <svg
    className="ax-icon ax-icon--dot"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<circle cx=\\"8\\" cy=\\"8\\" r=\\"6\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\"/>",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;

exports[`TextInputIconWrapper renders with defaultProps 1`] = `
<span
  className="ax-input__icon ax-input__icon--align-right"
>
  <svg
    className="ax-icon ax-icon--dot"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<circle cx=\\"8\\" cy=\\"8\\" r=\\"6\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\"/>",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;

exports[`TextInputIconWrapper renders with onClick 1`] = `
<span
  className="ax-input__icon ax-input__icon--align-right"
>
  <a
    className="ax-link ax-link--variant-normal"
    onClick={[Function]}
    style="subtle"
  >
    <svg
      className="ax-icon ax-icon--dot"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<circle cx=\\"8\\" cy=\\"8\\" r=\\"6\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\"/>",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </a>
</span>
`;

exports[`TextInputIconWrapper renders with tooltip when TooltipContext passed 1`] = `
<span
  className="ax-input__icon ax-input__icon--align-right"
>
  <svg
    className="ax-icon ax-icon--dot"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<circle cx=\\"8\\" cy=\\"8\\" r=\\"6\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\"/>",
      }
    }
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;

exports[`TextInputIconWrapper renders with tooltip when string passed 1`] = `
<span
  className="ax-input__icon ax-input__icon--align-right"
>
  <svg
    className="ax-icon ax-icon--dot"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<circle cx=\\"8\\" cy=\\"8\\" r=\\"6\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\"/>",
      }
    }
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;
